python/cudf/cudf/core/tools/datetimes.py

```diff
@@ -881,13 +900,12 @@ def date_range(
                 (res <= end) if _is_increment_sequence else (res <= start)
             ]
     else:
-        # If `offset` is fixed frequency, we treat both timestamps as integers
-        # and evenly divide the given integer range.
-        arr = cp.linspace(
-            start=start.value.astype("int64"),
-            stop=end_estim.astype("int64"),
-            num=periods,
-        )
+        # If `offset` is fixed frequency, we generate a range of
+        # treating `start`, `stop` and `step` as ints:
+        stop = end_estim.astype("int64")
+        start = start.value.astype("int64")
+        step = int(_offset_to_nanoseconds_lower_bound(offset))
+        arr = cp.arange(start=start, stop=stop, step=step)
```

Comment:
FYI @isVoid - let's sync up offline about these changes.

Reply:
I don't think this regression is tested in the test cases, should we add it? This snippet you sent me earlier:
```python
cudf.date_range('2001-01-01 00:00:00.000000', '2001-01-01 00:00:00.000010', freq='3us')
```
can be tested against pandas.